docs/guides/nodejs/survey-application.mdx

```diff
@@ -0,0 +1,369 @@
+  const { id, formId } = ctx.req.json()
+  const submission = await submissions.get(id)
+
+  if (!submission) {
+    console.error(`No submission found for ID: ${id}`)
+    return
+  }
+
+  // Build the PDF buffer from the submission data
+  const buffer = await buildReceipt(submission, formId)
+
+  // Store the PDF file in the bucket
+  const file = output.file(`${id}.pdf`)
+  await file.write(buffer)
+
+  console.log(`Receipt stored for ${id}`)
+})
+```
+
+### Build the PDF Receipt
+
+```ts title:form/receipt.ts
+import PDFDocument from 'pdfkit'
+import { feedbackSchema, rsvpSchema } from './schema'
+import { z } from 'zod'
+
+type FeedbackSubmission = z.infer<typeof feedbackSchema>
+type RsvpSubmission = z.infer<typeof rsvpSchema>
+
+export const buildReceipt = async (
+  data: any,
+  formId: string,
+): Promise<Buffer> => {
+  const receipt = new PDFDocument({ bufferPages: true })
+
+  const doneWriting = new Promise<Buffer>((resolve) => {
+    const buffers: Uint8Array[] = []
+
+    receipt.on('data', buffers.push.bind(buffers))
+    receipt.on('end', () => {
+      const pdfData = Buffer.concat(buffers)
+      resolve(pdfData)
+    })
+
+    receipt.font('Times-Roman').fontSize(20).text('Survey - Receipt', 100, 100)
+    receipt
+      .font('Times-Roman')
+      .fontSize(16)
+      .text('Submission Details', 100, 150)
+
+    if (formId === 'feedback') {
+      const { name, rating, feedback } = data as FeedbackSubmission
+
+      receipt
+        .font('Times-Roman')
+        .fontSize(12)
+        .text(
+          `Name: ${name}
+Rating: ${rating}
+Feedback: ${feedback || 'N/A'}`,
+          100,
+          175,
+        )
+    } else if (formId === 'rsvp') {
+      const { name, attending, guests } = data as RsvpSubmission
+
+      receipt
+        .font('Times-Roman')
+        .fontSize(12)
+        .text(
+          `Name: ${name}
+Attending: ${attending ? 'Yes' : 'No'}
+Guests: ${guests}`,
+          100,
+          175,
+        )
+    } else {
+      receipt
+        .font('Times-Roman')
+        .fontSize(12)
+        .text('Unknown form type. No details available.', 100, 175)
+    }
+
+    receipt.end()
+  })
+
+  return await doneWriting
+}
+```
+
+This PDF output is fairly plain, you can enhance it further using layout templates or branding.
+
+## Step 4: Delivery Logic
+
+Simulate or perform delivery of the receipt (e.g. via email or other downstream systems).
+
+```ts title:services/deliver.ts
+import { receipts } from '../resources/resources'
+
+receipts.subscribe(async (ctx) => {
+  const { id } = ctx.req.json()
+
+  // Simulate delivery or hook into a real email/SaaS integration
+  console.log(`Delivering receipt for submission: ${id}`)
+})
+```
+
+## Step 5: Retrieve Receipts
+
+Create an endpoint that returns a download URL for the generated receipt file.
+
+```ts title:services/receipts.ts
+import { api } from '@nitric/sdk'
+import { output } from '../resources/resources'
+
+const receiptApi = api('receipts')
```

Comment:
I'm assuming this isn't put in `resources/resources` because it's not shared.

Reply:
output is the bucket, i double chekced the code in examples repo and it looks to be exported.